plugins/services/src/js/structs/ServiceTree.js

```diff
@@ -99,6 +99,28 @@ module.exports = class ServiceTree extends Tree {
     });
   }
 
+  getItemParent(id, parent = null) {
+    if (this.getId() === id) {
+      return parent;
+    }
+
+    if (!this.list || this.list.length === 0) {
+      return null;
+    }
+
+    let parentFound = null;
+
+    this.list.forEach(child => {
+      if (child instanceof ServiceTree) {
+        parentFound = child.getItemParent(id, this);
+      } else if (child.getId() === id) {
+        parentFound = this;
```

Comment:
Instead of the fix on 114, I would also recommend to move away from forEach, and break/return on line 117.

It seems to me pointless to keep looking when we already found the item. I know it is premature optimization, but it is also more in line with a classic DFS.